protobufs/livekit_sip.proto

```diff
@@ -63,6 +63,7 @@ service SIP {
   rpc DeleteSIPTrunk(DeleteSIPTrunkRequest) returns (SIPTrunkInfo);
 
   rpc CreateSIPDispatchRule(CreateSIPDispatchRuleRequest) returns (SIPDispatchRuleInfo);
+  rpc GetSIPDispatchRule(GetSIPDispatchRuleRequest) returns (GetSIPDispatchRuleResponse);
```

Comment:
no preference here from me.. I think over time the list/get APIs might diverge (with pagination/etc)

Reply:
In general I prefer to have a separate get API, because it allows the service to send "not found" error directly, instead of requiring client checking the list and throwing that error.

But I see the point here. Reusing list API also allows seamless batching support, which is nice. Definitely worth it.

I rewrote the PR to add a few filters to list requests. The storage implementation won't need to support them all, because it will apply `Filter` after getting an initial list. Same for CLI / Go SDK - it will filter results after receiving them, so as soon as we update `protocol` in the client, it will immediately use new filter logic, even if the server doesn't support it yet.